packages/bot-config-utils/src/bot-config-utils.ts

```diff
@@ -45,16 +40,60 @@ function isFile(file: File | unknown): file is File {
   return (file as File).content !== undefined;
 }
 
+export interface ValidateConfigResult<T> {
+  isValid: boolean;
+  errorText: string | undefined;
+  config?: T;
+}
+
+export interface ValidateConfigOptions {
+  schema: object;
+  additionalSchemas?: Array<object>;
+}
+
+async function validateConfig<T>(
```

Comment:
Can it just be sync function? Maybe yes.
Also it is probably better idea to export this function. Other bot can utilize this for testing their schema.